jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -1629,21 +1632,67 @@ public VarHandle unreflectVarHandle(Field field) throws IllegalAccessException {
 			return targetClass;
 		}
 		
+		/*[IF Sidecar19-SE-B175]*/
 		/**
-		 * Return a class object with the same class loader, the same package
-		 * the same protection domain as the lookup class.
+		 * Return a class object with the same class loader, the same package and
+		 * the same protection domain as the lookup's lookup class.
 		 * 
 		 * @param classBytes the requested class bytes from a valid class file
 		 * @return a class object for the requested class bytes
 		 * @throws NullPointerException if the requested class bytes is null
-		 * @throws SecurityException if the SecurityManager prevents access
-		 * @throws LinkageError if the requested class is malformed, or fails in bytecode verification or linking
-		 * @throws IllegalAccessException if the access mode of the lookup doesn't include PACKAGE
 		 * @throws IllegalArgumentException if the requested class belongs to another package different from the lookup class
+		 * @throws IllegalAccessException if the access mode of the lookup doesn't include PACKAGE
+		 * @throws SecurityException if the SecurityManager prevents access
+		 * @throws LinkageError if the requested class is malformed, or is already defined, or fails in bytecode verification or linking
 		 */
-		public Class<?> defineClass(byte[] classBytes) throws NullPointerException, SecurityException, LinkageError, IllegalAccessException, IllegalArgumentException {
-			throw new UnsupportedOperationException("The method has not yet been implemented for now"); //$NON-NLS-1$
+		public Class<?> defineClass(byte[] classBytes) throws NullPointerException, IllegalAccessException, IllegalArgumentException, SecurityException, LinkageError {
+			if (Objects.isNull(classBytes)) {
+				/*[MSG "K065X", "The class byte array must not be null"]*/
+				throw new NullPointerException(com.ibm.oti.util.Msg.getString("K065X")); //$NON-NLS-1$
+			}
+			
+			/* Extract the package name from the class bytes so as to check whether
+			 * the requested class shares the same package as the lookup class.
+			 * Note: the class bytes should be considered as corrupted if any exception
+			 *       is captured in ASM class reader.
+			 */
+			ClassReader cr = null;
+			try {
+				cr = new ClassReader(classBytes);
+			} catch (ArrayIndexOutOfBoundsException e) {
+				throw new ClassFormatError(com.ibm.oti.util.Msg.getString("K065Y1")); //$NON-NLS-1$
```

Comment:
The external message K065Y1 needs to be declared.

Reply:
Already updated.